bundles/org.eclipse.orion.client.ui/web/edit/setup.js

```diff
@@ -1115,9 +1179,115 @@ objects.mixin(EditorViewer.prototype, {
 					editorViewer.updateDirtyIndicator();
 				});
 			}.bind(this));
+			this.editor.addEventListener("AnnotationModified", this.editorAnnotationModifiedListener = function(e) {
```

Comment:
The current logic of adding a breakpoint is: double click on line numbers -> send annotation event -> convert annotation to breakpoint -> change breakpoints -> receive breakpoint event -> convert breakpoint to annotation -> add annotation.
It's weird and not good for an embedded editor.
Any suggestion?